libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -184,7 +184,8 @@ - (SFRestAPISalesforceAction *)send:(SFRestRequest *)request delegate:(id<SFRest
     SFUserAccount *user = currentNetwork.account;
     if (user.credentials.accessToken == nil && user.credentials.refreshToken == nil && request.requiresAuthentication) {
         [self log:SFLogLevelInfo msg:@"No auth credentials found.  Authenticating before sending request."];
-        [[SFAuthenticationManager sharedManager] loginWithCompletion:^(SFOAuthInfo *authInfo) {
+        [[SFAuthenticationManager sharedManager] loginWithCompletion:^(SFOAuthInfo *authInfo, SFUserAccount *userAccount) {
+            [[SFUserAccountManager sharedInstance] setCurrentUser:userAccount];
```

Comment:
This doesn't seem right.  It seems like we should set `currentNetwork.account` with the updated account.  And there should probably be considerations for whether `currentNetwork` or `currentNetwork.account` are `nil` beforehand, and maybe disallow this action if so.  I.e. `SFRestAPI` should not be in the business of setting `currentUser`, even if it doesn't exist.  That's what `SalesforceSDKManager` is for.